File: programmers/연습문제/프린터.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    
    queue<pair<int, int>> q;
    priority_queue<int> pq;
    for (int i = 0; i < priorities.size(); i++) {
        int priority = priorities[i];
        q.push(make_pair(priority, i));
        pq.push(priority);
    }
    while (!q.empty()) {
        if (q.front().first == pq.top()) {
            answer++;
            if (q.front().second == location) {
                break;
            }
            pq.pop();
            q.pop();
        } else {
            pair<int, int> tmp = q.front();
            q.pop();
            q.push(tmp);
        }
    }
    
    return answer;
}

File: programmers/연습문제/level2/프린터.py
def solution(priorities, location):
    answer = 0
    q = [(v, i) for i, v in enumerate(priorities)]
    
    while True:
        item = q.pop(0)
        
        # if q and max(q)[0] > item[0]:
        if any(item[0] < temp[0] for temp in q):
            q.append(item)
        else:
            answer += 1
            if location == item[1]:
                return answer